### PR TITLE
Migrate to latest `darkside.proto` to help fix Darkside tests

### PR DESCRIPTION
**Pull request by str4d**
Depends on https://github.com/zcash/lightwalletd/pull/448.

This code review checklist is intended to serve as a starting point for the author and reviewer, although it may not be appropriate for all types of changes (e.g. fixing a spelling typo in documentation).  For more in-depth discussion of how we think about code review, please see [Code Review Guidelines](../blob/main/CODE_REVIEW_GUIDELINES.md).

# Author

- [ ] Self-review: Did you review your own code in GitHub's web interface? Code often looks different when reviewing the diff in a browser, making it easier to spot potential bugs.
- [ ] Automated tests: Did you add appropriate automated tests for any code changes?
- [ ] Code coverage: Did you check the code coverage report for the automated tests?  While we are not looking for perfect coverage, the tool can point out potential cases that have been missed.
- [ ] Documentation: Did you update Docs as appropiate? (E.g [README.md](../blob/main/README.md), etc.)
- [ ] Run the app: Did you run the app and try the changes? 
- [ ] Did you provide Screenshots of what the App looks like before and after your changes as part of the description of this PR? (only applicable to UI Changes)
- [ ] Rebase and squash: Did you pull in the latest changes from the main branch and squash your commits before assigning a reviewer? Having your code up to date and squashed will make it easier for others to review. Use best judgement when squashing commits, as some changes (such as refactoring) might be easier to review as a separate commit.

# Reviewer

- [ ] Checklist review: Did you go through the code with the [Code Review Guidelines](../blob/main/CODE_REVIEW_GUIDELINES.md) checklist?
- [ ] Ad hoc review: Did you perform an ad hoc review?  _In addition to a first pass using the code review guidelines, do a second pass using your best judgement and experience which may identify additional questions or comments. Research shows that code review is most effective when done in multiple passes, where reviewers look for different things through each pass._
- [ ] Automated tests: Did you review the automated tests?
- [ ] Manual tests: Did you review the manual tests?_You will find manual testing guidelines under our [manual testing section](../blob/mater/docs/testing/manual_testing)_
- [ ] How is Code Coverage affected by this PR? _We encourage you to compare coverage befor and after your changes and when possible, leave it in a better place. [Learn More...](../blob/master/docs/testing/local_coverage.md)_
- [ ] Documentation: Did you review Docs, [README.md](../blob/master/README.md), [LICENSE.md](../blob/master/LICENSE.md), and [Architecture.md](../blob/master/docs/Architecture.md) as appropriate?
- [ ] Run the app: Did you run the app and try the changes? While the CI server runs the app to look for build failures or crashes, humans running the app are more likely to notice unexpected log messages, UI inconsistencies, or bad output data.

### CODE REVIEW

File: Tests/DarksideTests/AdvancedReOrgTests.swift [hunk 1]
```diff
@@ -33,7 +33,7 @@ class AdvancedReOrgTests: ZcashTestCase {
             walletBirthday: birthday + 50,
             network: network
         )
-        try await coordinator.reset(saplingActivation: 663150, branchID: self.branchID, chainName: self.chainName)
+        try await coordinator.reset(saplingActivation: 663150, startSaplingTreeSize: 128607, startOrchardTreeSize: 0, branchID: self.branchID, chainName: self.chainName)
```
Comment (str4d) [inline Tests/DarksideTests/AdvancedReOrgTests.swift:36]: A bunch of the tests reference block height 663150, which corresponds to a persisted mainnet block elsewhere in the codebase. So everywhere I see that reference, I applied the Sapling tree size corresponding to it. There's likely a nicer way to do this, but that's refactoring I don't know how to do in Swift.

**Comment by str4d** [comment]
Rebased on current `main`. I think this PR can be reviewed and merged (as the corresponding `lightwalletd` PR has been), and then we can continue fixing the Darkside tests in subsequent PRs.

**Comment by LukasKorba** [comment]
I ran test yesterday evening with this result so I left it for today to re-check but ended up with the same result. 

Screenshot 2023-08-24 at 13 43 15

**Review by pacu — COMMENTED** [review]
first pass of review.

Will attempt to run the test and check their state

File: Tests/DarksideTests/AdvancedReOrgTests.swift (hunk 1 above)
Comment (nuttycom) [reply Tests/DarksideTests/AdvancedReOrgTests.swift:36]: Obtaining the tree states is what enables fast spendability; I believe that @str4d has already made the required change to lightwalletd here: https://github.com/zcash/lightwalletd/pull/448

Comment (nuttycom) [reply Tests/DarksideTests/AdvancedReOrgTests.swift:36]: That being said, the wallets should still function when connected to a LWD that doesn't support retrieval of tree states, but they'll be limited to linear scanning.

**Review by pacu — COMMENTED** [review]
I ran the tests using SDK `4c74e4d2041d5432bb25696b6bbf658b321fbf55` and LWD current `master` branch commit `8269810eeefac56ce326b0f6878c1ea3821b8ad5`.

The dLWD tests are able to start and reset the chain. 
image

The Tests fail for other reasons that appear not to be related to `darkside.proto` failures.
````
Sync failed with error: rustScanBlocks("Error while scanning blocks: The underlying block store produced the following error: Requested height 663199 does not exist in the block cache")
````

If this is the sole purpose of the PR as it appears to be, then I consider it approved.

but I can't approve per the Repo's policy
image

File: Tests/DarksideTests/AdvancedReOrgTests.swift (hunk 1 above)
Comment (str4d) [reply Tests/DarksideTests/AdvancedReOrgTests.swift:36]: @pacu I fetxhed the tree size for that block height from my mainmet zcashd node. One of the RPCs exposes it.

**Review by LukasKorba — APPROVED** [review]

**Pull request merged**